src/lib/rest/rest.cpp

```diff
@@ -974,46 +993,56 @@ static int connectionTreat
   //
   if (dataLen != 0)
   {
-    if (dataLen == ciP->httpHeaders.contentLength)
+    //
+    // If the HTTP header says the request is bigger than our PAYLOAD_MAX_SIZE,
+    // just silently "eat" the entire message
+    //
+    if (ciP->httpHeaders.contentLength > PAYLOAD_MAX_SIZE)
```

Comment:
Question: Is it necessary to read the whole body?  Could we send an error earlier? closing connection maybe ... (only asking)